DM-32944: Distinguish AuxTel cameras

jmeyers314 opened the pull request:
Previously had Instrument AuxTel as the ZWO camera
and nothing for LATISS.  Now, AuxTel is LATISS and
AuxTelZWO is the high speed ZWO camera.


Match the validated parameters from https://github.com/lsst-ts/ts_externalscripts/blob/develop/python/lsst/ts/externalscripts/auxtel/latiss_cwfs_align.py#L239

jmeyers314 commented:
Also, this PR may interfere with #95 , so keep that in mind when merging.

jmeyers314 commented:
Note that to get the pre-existing AuxTel unit tests to pass, I'm using what I'm confident is the wrong value for the ZWO pixel scale: `14.4e-6`.  From https://astronomy-imaging-camera.com/product/asi1600mm-cool, the ZWO camera has a pixel scale of `3.8e-6` and array shape of `4656x3520`.  The test images in `tests/testData/testImages/auxTelZWO/` (formerly `tests/testData/testImages/auxTel/`) are `1164x880` indicating a 4x binning.  `4x3.8` microns is `15.2` microns; I think the `14.4` micron above is a transcription error of `4x3.6` microns.

The right thing would be to use `4x3.8e-6 = 15.2e-6` in `policy/cwfs/instData/auxTelZWO/instParam.yaml`, and recompute the validation comparison values in `tests/testData/testImages/validation/auxTelZWO/`.  I wasn't able to reproduce these values though, so am unsure what the right thing to compare against is.

teweitsai commented:
If we do not use the `AuxTelZWO` anymore, please consider to replace the test data with LATISS, use the `danish` or `ts_wep` to calculate the wavefront error and put them as the test result for the comparison purpose. The unit test is to just protect the code for the future's refactoring. But I can approve this PR as well if you prefer to keep the original test data. Please just let me know your preference.

In addition, please help to update the version history file (`doc/versionHistory.rst`).

Thank you!

jmeyers314 commented:
Thanks @teweitsai .  For now I just updated the expected values for the AuxTelZWO test for the corrected pixel scale.  It probably would be a good idea to add some LATISS data to test against too, but I think that can be a separate issue/PR.  I updated the version history too.

teweitsai commented on doc/versionHistory.rst:
```diff
@@ -16,6 +16,7 @@ Version History
 * Update Jenkinsfile for CI job:
     * git command is no longer working after the latest update on our Jenkins server.
     * update path to plantuml.
+* Distinguish AuxTel ZWO camera from LATISS
```
Please consider to add the new version of `2.0.2`. Or do you want to retag this after merging the branch?

jmeyers314 replied:
Hi @teweitsai .  Sorry this has dragged on so long.  Hopefully you remember this comment from 6 weeks ago :)

I moved this comment to a new "version 2.2.1" section, which seems to be the current value after rebasing onto `develop`.  I haven't git tagged anything though.  I'm happy to add a tag if that's the culture for this group; please let me know.

teweitsai replied:
Sure. Please help to use the `v2.2.1` (depends on current condition) and `git tag` it after the merge. Thanks!

teweitsai commented on policy/cwfs/instData/auxTelZWO/instParam.yaml:
```diff
@@ -0,0 +1,24 @@
+
+# Auxiliary telescope parameters:
+
+# This is for the high speed ZWO camera
```
Please help to provide the full name of ZWO.

jmeyers314 replied:
I tried to google for this.  Best I could find is ZWO = "ZW Optical", but I still have no idea what ZW stands for.  Maybe "ZW Optical" is sufficient?

teweitsai replied:
ZW Optical is fine~ Thanks!

teweitsai reviewed (approved):
The update looks good to me. Thanks for this improvement!

teweitsai commented:
BTW, please help to update the description in `README.md` as well:
https://github.com/lsst-ts/ts_wep#note-of-auxiliary-telescope-images

Thank you!

suberlak commented on policy/cwfs/instData/auxTel/instParam.yaml:
```diff
@@ -2,6 +2,8 @@
+# This is for the LATISS camera
+
 # Obscuration (inner_radius / outer_radius of primary mirror)
 obscuration: 0.3525
```
During the AOS telecon we agreed that this should be `0.423` (`obscuration = inner_radius / outer_radius=0.2538/0.6=0.423`), also mentioned in https://lsstc.slack.com/archives/DGH6LJC7K/p1636059355007000  . Could this be updated before the merge ?

jmeyers314 replied:
Err...  I actually think it's currently correct.  Take a look here https://lsstc.slack.com/archives/C9BEJU1T3/p1638304311129800

So 

inner_diameter = 0.423
inner_radius = 0.2115
outer_radius = 0.6
obscuration = 0.3525

Let's see if @tribeiro and @patrickingraham can come to a consensus on this...
